### PR TITLE
Update vcpkg snapshot to 2021.05.12

### PR DESCRIPTION
**Pull request by j-rivero**
Update the vcpkg snapshot used by all our CI jobs to `2021-05-12`. 

@Blast545 was testing the different supported versions:
 * Citadel / Gazebo3: not supported
 * Dome / Gazebo4: not supported
 * Edifice / Gazebo5: https://build.osrfoundation.org/job/_vcpkg_testing_snapshot/104/
 * Fortress / Gazebo6: https://build.osrfoundation.org/job/_vcpkg_testing_snapshot/102/
 * Garden / GazeboMain: https://build.osrfoundation.org/job/_vcpkg_testing_snapshot/103/

The Windows machine should be update using https://build.osrfoundation.org/job/_vcpkg_update_snapshot/ before merging this.

### CODE REVIEW

**Review by Blast545 — APPROVED** [review]
LGTM, I'd keep an extra look once this is merged JIC.

**Review by chapulina — APPROVED** [review]

**Pull request merged**
